homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,233 @@
+"""Cover entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+import logging
+from typing import Any
+
+from motionblindsble.const import MotionBlindType, MotionRunningType
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN, ICON_VERTICAL_BLIND
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class MotionCoverEntityDescription(CoverEntityDescription):
+    """Entity description of a cover entity with default values."""
+
+    key: str = field(default=CoverDeviceClass.BLIND.value, init=False)
```

Comment:
```suggestion
    key: str = CoverDeviceClass.BLIND.value,
```

Why does this not work?

Reply:
If I remember correctly that is needed to override the default value, but is only needed for device_class so for the other two it can be removed I think
```
    key: str = CoverDeviceClass.BLIND.value
    translation_key: str = CoverDeviceClass.BLIND.value
    device_class: CoverDeviceClass = field(default=CoverDeviceClass.SHADE, init=True)
```

Reply:
I think you can do without

Reply:
I can remove it for now but can't test it so it may raise an error, but if that's the case I'll change it back and let you know

Do I even need to set the key and translation_key now that the extra state attribute is gone and the cover entity is no longer being translated?

Reply:
Or should I just keep the key and translation_key to the default value?

```python
@dataclass(frozen=True)
class MotionCoverEntityDescription(CoverEntityDescription):
    """Entity description of a cover entity with default values."""

    key: str = CoverDeviceClass.BLIND.value
    translation_key: str = CoverDeviceClass.BLIND.value


COVER_TYPES: dict[str, MotionCoverEntityDescription] = {
    MotionBlindType.HONEYCOMB.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.SHADE
    ),
    MotionBlindType.ROMAN.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.SHADE
    ),
    MotionBlindType.ROLLER.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.SHADE
    ),
    MotionBlindType.DOUBLE_ROLLER.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.SHADE
    ),
    MotionBlindType.VENETIAN.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.BLIND
    ),
    MotionBlindType.VENETIAN_TILT_ONLY.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.BLIND
    ),
    MotionBlindType.CURTAIN.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.CURTAIN
    ),
    MotionBlindType.VERTICAL.name: MotionCoverEntityDescription(
        device_class=CoverDeviceClass.CURTAIN, icon=ICON_VERTICAL_BLIND
    ),
}
```